### PR TITLE
proposal: Allow pvc.Selector and pvc.Class both to be set.

### PR DESCRIPTION
**Pull request by jsafrane**
To simplify initial implementation we made selector and class mutually exclusive. Now it's time to finish the implementation and let users to choose e.g. a zone where a volume should be provisioned in pvc.Selector.

Implementation is prepared in #37497 (work in progress, needs #38505 first)

@kubernetes/sig-storage-misc

### CODE REVIEW

**Comment by pospispa** [comment]
LGTM.

File: docs/proposals/volume-provisioning.md [hunk 1]
```diff
@@ -485,15 +485,55 @@ parameters:
    type: spinning
 ```
 
+### AWS provisioner and a PVC w
```
Comment (eparis) [inline docs/proposals/volume-provisioning.md]: `s/ w^//`

File: docs/proposals/volume-provisioning.md [hunk 2]
```diff
@@ -485,15 +485,55 @@ parameters:
...
+    "volume.beta.kubernetes.io/storage-class": "aws-fast"
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1500Mi
+  selector:
+    matchLabels:
+      "failure-domain.beta.kubernetes.io/zone": "us-east-1a"
```
Comment (eparis) [inline docs/proposals/volume-provisioning.md]: How do I know that my provisioner can handle `"failure-domain.beta.kubernetes.io/zone": "us-east-1a"` ?

I would love love love love love for the StorageClass to indicate the valid set of selectors. I really find this a useless feature without some describing...

Comment (gnufied) [inline docs/proposals/volume-provisioning.md]: On unrelated why are we calling the label key `failure-domain.beta.kubernetes.io/zone` ? Can we not keep it something simple like `zone`?  I understand that - pvc will get `failure-domain.beta.kubernetes.io/zone` label after it gets provisioned but I always understood `failure-domain.beta.kubernetes.io/zone` as a internal detail.

Comment (eparis) [inline docs/proposals/volume-provisioning.md]: I think having the label names namespaced is a good idea as an end user who makes their own label convention might want to use 'zone' to mean something totally different. Honestly having 'beta' in there completely sucks, but as painful as label namespacing is I think it is worth it to keep down any future confusion...

**Comment by jsafrane** [comment]
proposals have been moved to kubernetes/community repository, closing in favor of https://github.com/kubernetes/community/pull/247

File: docs/proposals/volume-provisioning.md (hunk 2 above)
Comment (jsafrane) [inline docs/proposals/volume-provisioning.md]: Name of the labels comes from https://github.com/kubernetes/kubernetes/blob/master/docs/api-reference/labels-annotations-taints.md which is already used e.g. by admission controller that adds this label to all PVs on GCE and AWS.

Comment (thockin) [inline docs/proposals/volume-provisioning.md]: eventually that needs to become un-beta

**Pull request closed without merging**